en/08_Changelogs/6.0.0.md

```diff
@@ -448,6 +449,19 @@ Note that the `DB::CONN_PRIMARY` constant, which has a value of "primary", is us
 
 Note that some [`DataQuery`](api:SilverStripe\ORM\DataQuery) methods such as [`DataQuery::execute()`](api:SilverStripe\ORM\DataQuery::execute()) now work slightly differently as they will use the replica database if the queried `DataObject` has the [`DataObject.must_use_primary_db`](api:SilverStripe\ORM\DataObject->must_use_primary_db) configuration set to `true`. However calling the equivalent [`SQLSelect`](api:SilverStripe\ORM\Queries\SQLSelect) method via a `DataQuery` e.g. `$dataQuery->query()->execute()` will not respect the `DataObject.must_use_primary_db` configuration.
 
+### Performance improvements for site tree rendering {#site-tree-performance}
+
+Performance improvements have been made to the rendering of the site tree used to show all of the pages in the CMS using a combination of more efficient database queries and in-memory caching. The primary focus of this work was to reduce the total number of database requests being made, as this has been found to be a significant source of perceived slowness when Silverstripe CMS is in a distributed hosting setup where the database is on a different physical server from the webserver.
+
+As part of this change a new method [`Hierarchy::getChildrenForTree()`](api:SilverStripe\ORM\Hierarchy\Hierarchy::getChildrenForTree()) has been created which replaces [`Hierarchy::AllChildrenIncludingDeleted()`](api:SilverStripe\ORM\Hierarchy\Hierarchy::AllChildrenIncludingDeleted()) which is now deprecated. However `AllChildrenIncludingDeleted()` includes an extension hook `augmentAllChildrenIncludingDeleted` which will no longer work unless you add configuration to use the old, slower method. This can be done with the following config:
```

Comment:
```suggestion
As part of this change a new method [`Hierarchy::getChildrenForTree()`](api:SilverStripe\ORM\Hierarchy\Hierarchy::getChildrenForTree()) has been created which replaces [`Hierarchy::AllChildrenIncludingDeleted()`](api:SilverStripe\ORM\Hierarchy\Hierarchy::AllChildrenIncludingDeleted()) as the default method used to get children to render in the site tree. `Hierarchy::AllChildrenIncludingDeleted()` is now deprecated. 

If you want to use `Hierarchy::AllChildrenIncludingDeleted()` to get items to display in the site tree (for example to take advantage of the`augmentAllChildrenIncludingDeleted` extension hook), this can be done with the following config:
```

Reply:
Updated